beacon-chain/sync/initial-sync/round_robin.go

```diff
@@ -304,7 +304,8 @@ func (s *Service) isProcessedBlock(ctx context.Context, blk block.SignedBeaconBl
 	if err != nil {
 		return false
 	}
-	if blk.Block().Slot() <= finalizedSlot || (s.cfg.DB.HasBlock(ctx, blkRoot) || s.cfg.Chain.HasInitSyncBlock(blkRoot)) {
+	blockExistsInDB := s.cfg.DB.HasBlock(ctx, blkRoot) || s.cfg.Chain.HasInitSyncBlock(blkRoot)
+	if blk.Block().Slot() <= finalizedSlot || blockExistsInDB && s.cfg.Chain.HeadSlot() >= blk.Block().Slot() {
```

Comment:
```suggestion
	if blk.Block().Slot() <= finalizedSlot || (s.cfg.DB.HasBlock(ctx, blkRoot) || s.cfg.Chain.HasInitSyncBlock(blkRoot)) && s.cfg.Chain.HeadSlot() >= blk.Block().Slot() {
```

This is kind of a nasty long line, but `blk.Block().Slot() <= finalizedSlot ` is true for most of initial sync. Putting this boolean first in the statement could reduce the number of db lookups.